components/metrics/lib.rs

```diff
@@ -61,22 +43,19 @@ fn set_metric<U: ProgressiveWebMetric>(
 ) {
     attr.set(Some(metric_time));
 
-    // Queue performance observer notification.
-    pwm.send_queued_constellation_msg(metric_type, metric_time);
```

Comment:
Before, the `set_metric` method when called for `ProgressiveWebMetricType::TimeToInteractive` would also queue the message to script thread. The handler for that message created the `PerformanceEntry` DOM object corresponding to the TTI metric as well.

But now  the caller of `set_metric`  is responsible for creating a `PerformanceEntry` DOM object. In the new code, only `handle_paint_metric` creates the `PerformanceEntry` object and it explictly only handles `FirstPaint` and `FirstContentfulPaint` metrics types, so we never create a `PerformanceEntry` DOM object in the new code. Isn't that a bug?

Reply:
I don't think this is a change in behavior. Note a couple of things:

1. The implementation of `send_queued_constellation_msg` for `InteractiveMetrics` is empty, so this line of code wasn't doing anything for the TTI metric.
2. The TTI metric isn't a `PerformancePaintTiming`, which is the specific type of `PerformanceEntry` that was created by this code path. In fact, TTI [isn't part of any web specification at all](https://developer.mozilla.org/en-US/docs/Glossary/Time_to_interactive), which I guess is why it doesn't have a corresponding DOM object.

That said, let me trigger tests to make sure this is the case.

Reply:
> The implementation of send_queued_constellation_msg for InteractiveMetrics is empty, so this line of code wasn't doing anything for the TTI metric.

Ah, I didn't realize that was an empty implementation. You are correct that the behaviour is not changed. 

> In fact, TTI [isn't part of any web specification at all](https://developer.mozilla.org/en-US/docs/Glossary/Time_to_interactive), which I guess is why it doesn't have a corresponding DOM object.

I see. I saw that there was an WebIDL interface defined [here](https://github.com/WICG/time-to-interactive?tab=readme-ov-file#api) and assumed that meant we wanted a DOM object. But I guess that was just a proposal and not part of the standard, so we don't have one right now.